libs/wingsdk/src/target-tf-azure/app.ts

```diff
@@ -61,5 +86,49 @@ export class App extends CdktfApp implements IApp {
       enumerable: false,
       writable: false,
     });
+
+    // At runtime the resolved app class is an instance of InnerApp which means that all the methods
+    // declared in this class are tossed out. This is a workaround to attach the getters to InnerApp class
+    // at runtime.
+    this._resourceGroup = undefined as any;
+    Object.defineProperty(this, "resourceGroup", {
```

Comment:
@eladb @staycoolcall911 @Chriscbr Wanted to draw your attention to this. Im still trying to understand why, but essentially because the instance of app is that of InnerApp from the CdktfApp during runtime creating a getter (or any other public method) gets tossed out. @MarkMcCulloh might have better explanation, but he rescued me on this one 😅. Though during our debugging we thought it might merit some more team discussion around this.

Reply:
Not sure I understand...

Is this what you tried to do and didn't work?

```ts
class Foo {
  private _resourceGroup?: ResourceGroup;

  public get resourceGroup() {
    if (!this._resourceGroup) {
      this._resourceGroup = new ResourceGroup(this, "ResourceGroup", ...);
    }

    return this._resourceGroup;
  }
}
```

Reply:
Thats right that getter method would never be called, because the only methods/fields available are those of the InnerApp class. Though what I dont understand is why when we create an instance of the app directly 
```ts
const app = new tfazure.App({ outdir: mkdtemp(), location: "East US" });
console.log(app);
```
this logged app is an InnerApp

Reply:
@hasanaburayyan @eladb Odd... actually I think I might have also encountered some issues like this while trying to set up stuff on the GCP bucket resource. I think it's possible this might be fixable by refactoring away the `CdktfApp` - take a look at what I did in my PR here (in core/app.ts):

https://github.com/winglang/wing/pull/827/files#diff-380d8cb14bea95361d57e8a63cc67a761a28a2e36214a025446ffef5f4c5adfe

I haven't cleaned up everything yet but I think with some tweaking it might be able to fix the issue.

Reply:
Yea I think refactoring out the InnerApp class is going to be the fix, once that PR is merged Ill get this rebased and fixed to be cleaner!

Reply:
Yupp, rebased the new changes to Cdktf class and things work the way they should 😁